Automattic/jetpack
# Use `composer update` rather than `install` when lockfile isn't checked in.

Partially fixes #18753

#### Changes proposed in this Pull Request:

* The CLI needs to detect whether the lockfile is checked in or not.
* It also has a bug where it wasn't doing the installs concurrently like it intended to.
* All the packages' `composer.json` should use `@composer update` in the scripts.

This PR doesn't fix the cli `generate` command's composer.json, as it currently doesn't seem to have logic determining whether composer.lock is checked in or not. Thus this doesn't completely fix #18753.

#### Jetpack product discussion


#18753

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Try `jetpack install` for projects with and without the lock files checked in, and for those without it checked in try it with or without the lock file existing. Ensure the correct command was run.
